update IBFT2.0 transitions

rolandtyler opened the pull request:
Signed-off-by: Roland Tyler <roland.tyler@consensys.net>

## Pull request checklist

## Describe the change
Edit transitions for IBFT2.0:
- update block period
- add block reward
<!-- A clear and concise description of what this PR changes in the documentation. -->

## Issue fixed
Fixes #843 
<!-- Except for minor changes (typos, commas) it's required to have a Github issue linked to your
pull request.

Use the following to make Github close the issue automatically when merging the PR:
fixes #{your issue number}
If multiple issues are involved, use one line for each issue.

If you don't want to close the issue, use:
see #{your issue number} -->

## Impacted parts <!-- check as many boxes as needed -->

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

## Testing
https://hyperledger-besu--882.org.readthedocs.build/en/882/HowTo/Configure/Consensus-Protocols/IBFT/
<!-- Steps to follow to review and test your changes.
Add links to preview the pages changes here.
Link format is https://hyperledger-besu--{your PR number}.org.readthedocs.build/en/{your PR number}/
Where {your PR number} must be replaced by the number of this PR, for instance 123
-->

siladu commented on docs/HowTo/Configure/Consensus-Protocols/IBFT.md:
```diff
@@ -181,7 +180,22 @@ To tune the block timeout for your network deployment:
+
+{!global/Config-Options.md!}
+
+## Transitions
```
I like that Transitions have their own section now, that'll be helpful for linking to it which I seem to be doing more often these days :)

siladu commented on docs/HowTo/Configure/Consensus-Protocols/IBFT.md:
```diff
@@ -245,12 +259,87 @@ To update an existing network with a new `blockperiodseconds`:
+                 "ibft2": {
+                   "blockperiodseconds": 2,
+                   "epochlength": 30000,
+                   "requesttimeoutseconds": 4
```
Maybe should have an initial blockreward value here to show what its being changed in the transition below?

siladu replied:
It defaults to zero if you don't specify it in the initial config, so the example is still correct. If the intention is to go from 0 -> +ve reward then IMO we should make that explicit either in the preamble or by specifying initial config.

siladu commented on docs/HowTo/Configure/Consensus-Protocols/IBFT.md:
```diff
@@ -245,12 +259,87 @@ To update an existing network with a new `blockperiodseconds`:
+                 "ibft2": {
+                   "blockperiodseconds": 2,
+                   "epochlength": 30000,
+                   "requesttimeoutseconds": 4
```
As above, initial blockreward would be good here, unless the example intends to show 0 -> +ve value

siladu commented on docs/HowTo/Configure/Consensus-Protocols/IBFT.md:
```diff
@@ -245,12 +259,87 @@ To update an existing network with a new `blockperiodseconds`:
+                   "ibft2": [
+                   {
+                     "block": 10,
+                     "blockreward": "1"
```
The blockreward unit is Wei. If the aim of these examples is to show realistic number, we could probably add quite a few more 0s on the end :)

siladu replied:
e.g. https://besu.hyperledger.org/en/stable/HowTo/Configure/Consensus-Protocols/IBFT/#genesis-file uses "5000000000000000" Wei

siladu commented on docs/HowTo/Use-Privacy/Performance-Best-Practices.md:
```diff
@@ -14,6 +14,7 @@ However the following item is of such outstanding impact to performance that it
 
 Running EVM transactions creates a lot of random reads that are executed sequentially.
 The Besu data folder should be located on the fastest possible storage media.
+
```
errant space?

siladu replied:
ah...just realised that was the markdown fix 🏅

siladu reviewed (changes requested):
blockreward transitions apply to QBFT as well so it could be added to here https://besu.hyperledger.org/en/latest/HowTo/Configure/Consensus-Protocols/QBFT/#transitions
Not sure if that's beyond the scope of this PR though :)

A few minor points in the comments...

rolandtyler commented:
> blockreward transitions apply to QBFT as well so it could be added to here https://besu.hyperledger.org/en/latest/HowTo/Configure/Consensus-Protocols/QBFT/#transitions Not sure if that's beyond the scope of this PR though :)
> 
> A few minor points in the comments...

@siladu, thanks for the feedback.  I updated the examples and added the QBFT transition, as well as `miningbeneficiary`.    Could you re-review? Thanks!

siladu commented on docs/HowTo/Configure/Consensus-Protocols/IBFT.md:
```diff
@@ -241,16 +253,98 @@ To update an existing network with a new `blockperiodseconds`:
    [`ibft_getValidatorsByBlockNumber`](../../../Reference/API-Methods.md#ibft_getvalidatorsbyblocknumber), specifying `latest`.
 
-!!! caution
+### Update block rewards on an existing network deployment
```
The block period uses the title "Configure block time...". Should these be consisent, i.e. either both use "Configure..." or both use "Update..."?

siladu commented on docs/HowTo/Configure/Consensus-Protocols/QBFT.md:
```diff
@@ -405,6 +402,95 @@ To update an existing network with a new `blockperiodseconds`:
 4. To verify the changes after the transition block, call
    [`qbft_getValidatorsByBlockNumber`](../../../Reference/API-Methods.md#ibft_getvalidatorsbyblocknumber), specifying `latest`.
 
+### Update block rewards on an existing network deployment
```
Same point about "Configure..." and "Update..."

siladu reviewed (commented):
Nice one for adding to QBFT as well :)

I'm not sure why there are changes to the `miningbeneficiary` ...I don't believe it's possible to update that one, but apologies if I missed something - where did that requirement come from?

siladu reviewed (approved):
LGTM :)

bgravenorst commented on docs/HowTo/Configure/Consensus-Protocols/IBFT.md:
```diff
@@ -241,16 +255,93 @@ To update an existing network with a new `blockperiodseconds`:
 
-1. Restart all nodes in the network using the updated genesis file.
-1. To verify the changes after the transition block, call
+3. Restart all nodes in the network using the updated genesis file.
```
@rolandtyler these get automatically numbered if you leave all the ordered list items as `1.`

rolandtyler replied:
The version of IntelliJ I have has some auto feature that automatically renumbers them when I enter them as `1`.  There's still no release yet to fix it.  [Issue link](https://youtrack.jetbrains.com/issue/IDEA-273049).

bgravenorst reviewed (approved):
Just the one comment. otherwise, LGTM.
